### PR TITLE
Add support for provisioned users API endpoint

### PR DESCRIPTION
**Pull request by sfang97**
[In 14.8, GitLab added](https://gitlab.com/gitlab-org/gitlab/-/merge_requests/79600/) a new provisioned users API endpoint. The new endpoint allows group maintainers to list [users provisioned by a group](https://docs.gitlab.com/ee/user/group/saml_sso/scim_setup.html). 

[Provisioned users API documentation](https://docs.gitlab.com/ee/api/groups.html#provisioned-users-api)

### CODE REVIEW

**Comment by sfang97** [comment]
@svanharmelen 👋 Could you please review this when you have a chance? Thanks! 🙏

File: groups.go [hunk 1]
```diff
@@ -803,3 +803,39 @@ func (s *GroupsService) DeleteGroupPushRule(gid interface{}, options ...RequestO
 
 	return s.client.Do(req, nil)
 }
+
+type ListProvisionedUsersOptions struct {
+	ListOptions
+	Username             *string    `url:"username,omitempty" json:"username,omitempty"`
```
Comment (stanhu) [inline groups.go:809]: I think you need to run `go fmt` on this?

```diff
diff --git a/groups.go b/groups.go
index 3d24304..d819e54 100644
--- a/groups.go
+++ b/groups.go
@@ -806,12 +806,12 @@ func (s *GroupsService) DeleteGroupPushRule(gid interface{}, options ...RequestO
 
 type ListProvisionedUsersOptions struct {
 	ListOptions
-	Username             *string    `url:"username,omitempty" json:"username,omitempty"`
-	Search               *string    `url:"search,omitempty" json:"search,omitempty"`
-	Active               *bool `url:"active,omitempty" json:"active,omitempty"`
-	Blocked              *bool `url:"blocked,omitempty" json:"blocked,omitempty"`
-	CreatedAfter         *time.Time `url:"created_after,omitempty" json:"created_after,omitempty"`
-	CreatedBefore        *time.Time `url:"created_before,omitempty" json:"created_before,omitempty"`
+	Username      *string    `url:"username,omitempty" json:"username,omitempty"`
+	Search        *string    `url:"search,omitempty" json:"search,omitempty"`
+	Active        *bool      `url:"active,omitempty" json:"active,omitempty"`
+	Blocked       *bool      `url:"blocked,omitempty" json:"blocked,omitempty"`
+	CreatedAfter  *time.Time `url:"created_after,omitempty" json:"created_after,omitempty"`
+	CreatedBefore *time.Time `url:"created_before,omitempty" json:"created_before,omitempty"`
 }
 
 // ListProvisionedUsers gets a list of users provisioned by the given group.
```

File: groups.go [hunk 2]
```diff
@@ -803,3 +803,39 @@ func (s *GroupsService) DeleteGroupPushRule(gid interface{}, options ...RequestO
...
+// GitLab API docs:
+// https://docs.gitlab.com/ee/api/groups.html#provisioned-users-api
+func (s *GroupsService) ListProvisionedUsers(gid interface{}, opt *ListProvisionedUsersOptions, options ...RequestOptionFunc) ([]*User, *Response, error) {
+	group, err := parseID(gid)
+	if err != nil {
+		return nil, nil, err
+	}
+	u := fmt.Sprintf("groups/%s/provisioned_users", PathEscape(group))
+
+	req, err := s.client.NewRequest("GET", u, opt, options)
```
Comment (jaimem88) [inline groups.go:829]: You can use a predefined constant here. It also helps with consistency in other methods 😄 

```suggestion
	req, err := s.client.NewRequest(http.MethodGet, u, opt, options)
```

File: groups.go (hunk 1 above)
Comment (sfang97) [reply groups.go]: Thanks!

File: groups.go (hunk 2 above)
Comment (sfang97) [reply groups.go]: Thanks, applied!

**Comment by pschwartz** [comment]
@sfang97 Thank you for working on this so quickly.

@svanharmelen I have tested this against the new API as T-Mobile has direct need of it. Code works as intended.

**Comment by pschwartz** [comment]
@svanharmelen Any chance we can get a final review and merge on this?

**Review by svanharmelen — APPROVED** [review]

**Pull request merged**